src/internal/connector/sharepoint/api/pages.go

```diff
@@ -97,18 +97,21 @@ func FetchPages(ctx context.Context, bs *discover.BetaService, siteID string) ([
 			return err
 		})
 		if err != nil {
-			return nil, support.ConnectorStackErrorTraceWrap(err, "failed fetching site page")
+			return nil, clues.Wrap(err, "fetching site page").WithClues(ctx).WithAll(graph.ErrData(err)...)
 		}
 
 		for _, entry := range resp.GetValue() {
-			pid := *entry.GetId()
-			temp := Tuple{pid, pid}
-
-			if entry.GetName() != nil {
-				temp.Name = *entry.GetName()
+			var (
+				pid  = *entry.GetId()
+				temp = NameID{pid, pid}
+			)
+
+			name, ok := ptr.ValOK(entry.GetName())
```

Comment:
I'm not sure if `ValOK` improves the ergonomics here by much. I personally feel like previous approach of checking for nil was a bit more obvious to a person reading the code.

Reply:
Two way door; lets bring it up in weekly discussion.  Personally, I think it matches the map lookup format nicely.  Eg `v, ok := myMap[key]` vs `v, ok := ptr.ValOK(ptr)`.  (wouldn't it be nice to be able to optionally extend return values in normal funcs, for cases like this?)